src/components/modals/Basic/BasicModal.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { FC, HTMLProps, MouseEventHandler } from 'react'
+
+import styles from './basicModal.module.css'
+
+interface BasicModalCustomProps {
+   open?: boolean
+   onClose?: () => void
+}
+export type BasicModalProps = BasicModalCustomProps & HTMLProps<HTMLDivElement>
+
+const BasicModal: FC<BasicModalProps> = ({ className = '', open = false, onClose = () => {}, ...props }) => {
+   const preparedClasses = [styles.modal, className].join(' ')
+
+   const handleClose: MouseEventHandler<HTMLDivElement> = e => {
+      onClose()
+   }
+   const handleChildClick: MouseEventHandler<HTMLDivElement> = e => {
+      e.stopPropagation()
+   }
+   if (!open) return null
+
+   return (
+      <div className={styles.blackout} onClick={handleClose}>
+         <div data-testid="modal" {...props} className={preparedClasses} onClick={handleChildClick}></div>
+      </div>
+   )
```

Comment:
`createPortal` нужно вызывать внутри модалки, а не в вместе с компонентой. Как я понял, у тебя нужно будет делать так
```
createPortal(<BasicModal>...</BasicModal>, document.body)
```
Должно быть так
```suggestion
  createPortal(
      <div className={styles.blackout} onClick={handleClose}>
         <div data-testid="modal" {...props} className={preparedClasses} onClick={handleChildClick}></div>
      </div>
   , document.body)
```